zerver/tests/test_subs.py

```diff
@@ -1222,7 +1224,8 @@ def test_basic_add_delete(self) -> None:
 
         # now delete the same stream
         request = {
-            'delete': ujson.dumps(['my_test_stream_1'])
+            'delete': ujson.dumps(['my_test_stream_1']),
+            'principals': ujson.dumps([self.example_email("hamlet")])
```

Comment:
I am not sure, why I have to pass `principals` here even when deleting stream. If I remove principals it is throwing error `Stream must have at least one subscriber`. Try to investigate, but couldn't find reason.
User should not need `principals` while deleting stream. But then why it's throwing error of code `add_subscriptions_backend` in `stream.py`
We should definitely not throw this error when user is deleting stream and does not pass `principals`

Reply:
@YJDave: I don't think this is the right way to fix this test failure. I would recommend reversing the changes that pass in `principals` to all of these tests. I'll explain more in my comment below. :)